app/annotationsList/annotationList.js

```diff
@@ -75,11 +75,21 @@ app.controller('AnnotationListCtrl', function($rootScope, $scope, $http, $uibMod
   }
 
   function postProcess() {
```

Comment:
Data binding + promises should be async already. I tested by adding a delay to the taxonomy retrieval and only that column was delayed, the rest of the table came as fast as usual.